### PR TITLE
Update default infrastructure command to be set at runtime

### PR DESCRIPTION
**Pull request by madkinsz**
Originally, this command was set as a default on infrastructure so that when a deployment is created the command is frozen. This is helpful for containerized deployments: if we ever change the entrypoint command, the agent needs to send the correct command to the container. In the frozen implementation, we do not need to worry about any version tracking to achieve this. However, there are a few issues introduced by this:

- If the infrastructure is a subprocess and the entrypoint changes, the agent would use the wrong command
- Maintaining the `prefect.engine` entrypoint in the future is very low cost
- Setting the command to call `python` statically can cause flow runs to run outside the agent's virtual environment
- Infrastructure blocks can be used for running arbitrary commands, not just flow runs, but the default command was flow run specific
 
Since every other flow run related property is set when a flow run is submitted to infrastructure by an agent, the inclusion of this on the infrastructure itself is confusing. This pull request:

- Restores the type of `command` to `Optional[List[str]]`
- Eliminates language around flow run commands for infrastructure fields
- Uses `sys.executable` for processes to ensure agents submit flows to the correct Python version and environment
- Retains the ability to override the command
- Clarifies any confusion about whether or not the command has been provided by the user or should be the default
- Includes some minor changes to the descriptions of other base infrastructure fields
- Refactors infrastructure preparation to be a property of infrastructure
- Refactors the agent to prepare infrastructure for the flow run and call `run` directly

The refactor to move submission logic to the base infrastructure type was motivated by concerns raised by @cicdw where we needed to perform conditional logic based on infrastructure type to determine the flow run entrypoint command to use. This refactor removes the `submission` module entirely, adding a `prepare_for_flow_run` method to `Infrastructure` and flow run submission utilities as static methods on the infrastructure base class. 

Closes https://github.com/PrefectHQ/prefect/issues/6461

Remaining work:
- [x] Update tests to account for change
- [x] Add checks for null commands at infrastructure runtime
- [ ] Ensure this is backwards compatible 

### Checklist

- [x] This pull request references any related issue by including "closes `<link to issue>`"
	- If no issue exists and your change is not a small fix, please [create an issue](https://github.com/PrefectHQ/prefect/issues/new/choose) first.
- [x] This pull request includes tests or only affects documentation.
- [x] This pull request includes a `bug`, `feature`, `enhancement`, `docs`, or `maintenance` label categorizing the change.

### CODE REVIEW

**Comment by madkinsz** [comment]
Note that @cicdw and I chatted briefly about this and hope to think about the interaction between infrastructure, agents, and flow runs holistically. An alternative design is likely to improve clarity in the long run. For now, I think that consolidating the override behavior to submission by the agent simplifies this pattern.

**Comment by madkinsz** [comment]
One option for a more holistic approach is to allow `command` to be passed to `run`. The `command` field would then just be the _default_ command for the infrastructure.

**Review by bunchesofdonald — APPROVED** [review]
This makes sense to me. It seems best to have that base block agnostic about the eventual command that's going to run. I _do_ like the sound of `run` also taking a command, but that could just be my docker fanboy kicking in.

File: src/prefect/infrastructure/submission.py [hunk 1]
```diff
@@ -1,14 +1,23 @@
...
 from prefect.orion.schemas.core import FlowRun
 
 MIN_COMPAT_PREFECT_VERSION = "2.0b12"
 
 
+def get_flow_run_entrypoint(infrastructure):
+    if isinstance(infrastructure, Process):
+        return [sys.executable, "-m", "prefect.engine"]
+    else:
+        return ["python", "-m", "prefect.engine"]
```
Comment (cicdw) [inline src/prefect/infrastructure/submission.py:18]: The design of this feels off to me - the location of this command specification is so far removed from where a developer will look when debugging

Comment (cicdw) [reply src/prefect/infrastructure/submission.py:18]: Sad differently, it's far too implicit

Comment (madkinsz) [reply src/prefect/infrastructure/submission.py:18]: This single module deals with all of the overrides for running flows in infrastructure. I'm hoping that consolidation here makes it easier to find when debugging in contrast to v1 where you need to look through the implementation details of each agent. I'm not excited about the specific toggle for the Python call in process infrastructure, but I'm not sure where else to implement it. Without the `sys.executable` switch we're too likely to call into the wrong `python` for subprocesses.

Comment (madkinsz) [reply src/prefect/infrastructure/submission.py:18]: Are you okay with improving the design in the future? Which user stories are you worried this degrades?

Comment (madkinsz) [reply src/prefect/infrastructure/submission.py]: This was addressed with a refactor agreed upon in a team review session 🥳

**Review by cicdw — APPROVED** [review]
Nice, this looks clean!

**Review by cicdw — APPROVED** [review]

**Pull request merged**